bindings/mysql/mysql.go

```diff
@@ -112,23 +116,29 @@ func (m *Mysql) Init(ctx context.Context, md bindings.Metadata) error {
 		return fmt.Errorf("missing MySql connection string")
 	}
 
-	db, err := initDB(meta.URL, meta.PemPath)
+	m.db, err = initDB(meta.URL, meta.PemPath)
 	if err != nil {
 		return err
 	}
 
-	db.SetMaxIdleConns(meta.MaxIdleConns)
-	db.SetMaxOpenConns(meta.MaxOpenConns)
-	db.SetConnMaxIdleTime(meta.ConnMaxIdleTime)
-	db.SetConnMaxLifetime(meta.ConnMaxLifetime)
+	if meta.MaxIdleConns > 0 {
```

Comment:
possibly a behavior change here right? Not sure what setting the value to 0 (for unused metadata) would have done in all of those cases.

Reply:
The behavior now is correct - but unclear if something is changing for users or not.

Reply:
From the APIs:

![image](https://github.com/dapr/components-contrib/assets/43508/b11ac72f-f8e3-4fd3-895b-c612e74b0c76)

![image](https://github.com/dapr/components-contrib/assets/43508/62ce1ba5-2ad2-450c-aa5f-39c8e6b721f5)

![image](https://github.com/dapr/components-contrib/assets/43508/f84eb7c5-eea4-40a6-a965-50ec8e32b6ec)